### ISSUE TITLE
Pubmatic adapter should use site.content.data instead of private dctr param

### ISSUE THREAD

**Issue by bretg** [prebid/Prebid.js#9142]
## Type of issue
efficiency

## Description
We found that there can be many hundreds of Seller-Defined Audience segments defined in a request in site.content.data.

Further, we noticed that pubmatic is copying all of them to a bidder-specific dctr parameter before sending to Prebid Server

```
  transformBidParams: function (params, isOpenRtb, adUnit, bidRequests) {
   // wow - this doubles the size of an already XXXXXXL dataset.
    _addJWPlayerSegmentData(params, adUnit.bids[0], true);
    return convertTypes({
      'publisherId': 'string',
      'adSlot': 'string'
    }, params);
  }
};
```

Here's an example:
```
imp[0].ext.prebid.bidder:{
                        "pubmatic": {
                            "publisherId": "1111111",
                            "dctr": "iris_context=ic_1670398,ic_9534888,ic_3129871,ic_8322846,ic_4260700,ic_4962242,ic_9954675,ic_2592227,ic_8121076,ic_2001389,ic_0381735,ic_1592380,ic_2177941,ic_2416407,ic_6703731,ic_7753435,ic_1612662,ic_8328276,ic_0344266|jw-id=jw_NkSk6w4a|jw-80011026=1|jw-80011062=1|jw-80011064=1|jw-80011065=1|jw-80011079=1|jw-80011080=1|jw-80011081=1|jw-80011082=1|jw-CMTPI-001553FA=1|jw-CMTPI-0059DF08=1|jw-CMTPI-0081098E=1|jw-CMTPI-0142FEF1=1|jw-CMTPI-022AFFD7=1|jw-CMTPI-02947E6F=1|jw-CMTPI-03654CCA=1| ... 25KB more data ...
   ...
site: {
        "content": {
            "data": [
                {
                    "name": "jwplayer.com",
                    "segment": [
                        {
                            "id": "80011026",
                            "value": "80011026"
                        },
                        {
                            "id": "80011062",
                            "value": "80011062"
                        },
                        {
                            "id": "80011064",
                            "value": "80011064"
                        },
                        {
                            "id": "80011065",
                            "value": "80011065"
                        },
                        {
                            "id": "80011079",
                            "value": "80011079"
                        },
                        {
                            "id": "80011080",
                            "value": "80011080"
                        },
                        {
                            "id": "80011081",
                            "value": "80011081"
                        },
                        {
                            "id": "80011082",
                            "value": "80011082"
                        },
                        {
                            "id": "CMTPI-001553FA",
                            "value": "CMTPI-001553FA"
                        },
                        {
                            "id": "CMTPI-0059DF08",
                            "value": "CMTPI-0059DF08"
                        },
         ... 25KB more ...
```

We would like Pubmatic to become more efficient here:

1) update your server-side adapter so it can read the data from site.content.data
2) then update your client-side adapter so it's duplicating the data

thanks

Heads up: @pm-nitin-shirsat , @pm-azhar-mulla , @pm-harshad-mane, @pm-nilesh-chate, @sachin-pubmatic

**Comment by bretg** [comment]
FYI - tangentially related to #9141

**Comment by patmmccann** [comment]
also tagging @jlquaccia 

Aside: @jlquaccia we'd love to draft you as a reviewer. We very much appreciate your recent contributions.

**Comment by jlquaccia** [comment]
@patmmccann thanks! sounds good to me.  Also, thanks for the heads up on this gh issue.  I should be able to take a look soon sometime next week

**Comment by jlquaccia** [comment]
hey @ramkrishan-ravi, @deepak-dhage, @pm-nitin-shirsat, @pm-azhar-mulla , @pm-nilesh-chate, @sachin-pubmatic

Jumping back into this, wanted to put this ticket back on the radar (as it got a bit ice-boxed).  PR with frontend changes for this ticket has been ready, can one of you please take on the remainder of work for this ticket?

Need to know if merging https://github.com/prebid/Prebid.js/pull/9169 will have an impact on translator and if so could we address any necessary changes?  Know there was some movement on prior discussion before on the potential impact risks of merging 9169.

**Comment by patmmccann** [comment]
@jlquaccia Cafemedia may need to run a version of prebid with 9169 despite any revenue loss to Pubmatic, we cannot continue to wait

**Comment by jlquaccia** [comment]
@patmmccann really sorry for the wait on this one, spoke internally, we will make the necessary changes on our end.  In the meantime you are good to go with https://github.com/prebid/Prebid.js/pull/9169, it can be merged.

**Comment by patmmccann** [comment]
Thanks!

**Comment by patmmccann** [comment]
fixed on #9169
